src/navigator/root-stack.tsx

```diff
@@ -39,11 +42,17 @@ export const RootStackScreen = () => {
   const { isLocked } = useAppLock();
   const isAuthorised = useIsAuthorisedSelector();
   const colors = useColors();
+  const dispatch = useDispatch();
 
   const [currentRouteName, setCurrentRouteName] = useState<ScreensEnum>(ScreensEnum.Welcome);
 
   useQuickActions();
 
+  useEffect(() => {
+    dispatch(reinstallAction.submit);
+    dispatch(setIsReinstalled(true));
```

Comment:
possible race condition problem when `setIsReinstalled` reducer will change state faster than `reinstallAction.submit` epic will do the things.  
build straight logic (names are approximate):
1. `isFirstAppLaunchCheckAction` dispatch
2. First step action epic
2.1 filter by state flag (`isFirstAppLaunch` for example)
2.2 if `true` (default value) return `rootStateResetAction.submit` and action that will change the flag from step 2.1